### PR TITLE
Add RGB invoice chain validation

### LINKED ISSUE #360: Add chain / network to RGB invoice
From what I can tell, what chain / network the wallet is currently set is not being set nor is it validated on RGB invoices.

We should both set it, since there's a field for it, and validate it upon all methods that accept invoices.

### PR DESCRIPTION
**Pull request by oacdutra**
Closes #360

### CODE REVIEW

File: src/validators.rs [hunk 1]
```diff
@@ -43,12 +50,22 @@ impl Default for RGBContext {
     fn default() -> Self {
         Self {
             closed_method: "tapret1st".to_string(),
+            current_network: String::new(),
```
Comment (cryptoquick) [inline src/validators.rs:53]: Would it make sense to default to this being Option::None, since that's the type on RgbInvoice ?

**Review by cryptoquick — CHANGES_REQUESTED** [review]
Very fast, very good work! Just one question.

**Review by josediegorobles — APPROVED** [review]
ACK

File: src/validators.rs (hunk 1 above)
Comment (oacdutra) [reply src/validators.rs:53]: Yes, we can work with option. I will change.

**Review by cryptoquick — APPROVED** [review]
ACK

**Pull request merged**